components/builder-api/src/server/mod.rs

```diff
@@ -39,9 +40,15 @@ impl HttpGateway for ApiSrv {
 
     fn add_middleware(config: Arc<Self::Config>, chain: &mut iron::Chain) {
         chain.link(persistent::Read::<Self::Config>::both(config.clone()));
-        chain.link(persistent::Read::<GitHubCli>::both(
-            GitHubClient::new(config.github.clone()),
-        ));
+        if config.github.is_some() {
+            let c = config.github.clone().unwrap();
+            chain.link(persistent::Read::<GitHubCli>::both(GitHubClient::new(c)));
+        } else if config.bitbucket.is_some() {
+            let c = config.bitbucket.clone().unwrap();
+            chain.link(persistent::Read::<BitbucketCli>::both(
+                BitbucketClient::new(c),
+            ));
```

Comment:
May be good to do some validation here - eg, panic if neither or both are present/enabled?